### ISSUE TITLE
Add schedule filters for volunteer's day/time preferences

### ISSUE THREAD

**Issue by mmcneill** [redcross/arcdata#83]
This issue is part of a broader set of features related to improving the shift scheduling process. See docs/DCSOps_Functional_Specification.pdf  (#81) for background information and details.

When viewing a schedule, volunteers should be able to filter visible shifts by day/time preferences. 

![req2_1](https://cloud.githubusercontent.com/assets/5384790/13828889/028b9bb2-eb92-11e5-9b14-433ca507a501.png)

These scheduling preferences should automatically save when changed by the volunteer.  Each time the volunteer logs in to DCSOps and views the filtered calendar, the saved preferences from their last session should repopulate the check boxes and filter the schedule.  These schedule preferences are linked to a specific region’s calendar.  

On the back end, the only additional requirement for implementing this features is the creation of a volunteer_schedule_preferences table that stores scheduling preferences on a “one volunteer/region pair per row” basis (example below).

![req2_2](https://cloud.githubusercontent.com/assets/5384790/13828858/caa565ac-eb91-11e5-8feb-9847bce97d6f.png)

Each time a volunteer changes their scheduling preferences via the check box screen show above, the corresponding volunteer/region row in this table is updated or added.  Note: the “All Regions” tab is treated as its own separate region in this context.

**Comment by cecilia-donnelly** [comment]
This is very closely related to #82.  Wherever the changes are made for that issue, these should also be made.

Getting into the implementation details here, but we could also flip that table design on its side and do something like this:
```
| ID | Shift Name  | Region ID | Person ID | IsPreferred |
--------------------------------------------------------
|  1  | Mon.Shift1 |     COOK |      2345 |          true |
```


Are the shift names already listed/documented somewhere, @mmcneill?  Are there 6 shifts per day, as shown in the mock up?

**Comment by mmcneill** [comment]
@cecilia-donnelly I am not 100% sure, but from what I have seen of the data I don't think those six shifts are officially listed/named anywhere.  A shift is just a start time and an end time.  However, the website only allows most volunteers to choose from those six time blocks when they put themselves on the schedule so most of the shifts are one of those six.  

The only DAT role that has shifts not from the list of six shown in the mock up is DAT Captain (they have 6-hr shifts).  I am not sure how DAT Captains sign up for shifts (DCSOps or not).  

In short though, I suppose it would be better to "name" a shift in the data table by day of week, start time, and end time instead of something like Mon.Shift1 in case the shift/preference timing options change in the future (and just to avoid confusion).

**Comment by jasonwinget** [comment]
There are different patterns of shifts.  Some are a week at a time, some are a day at a time, some are 2 hours and many are something inconsistently in between.  Every shift can be configured differently.  There is no specific set pattern.

**Comment by mmcneill** [comment]
@jasonwinget good to know, thanks.  I should note that the data I have access to is all from the American Red Cross of Greater Chicago so some of my observations will only apply to chapters in Northern Illinois and Northwest Indiana.  Here (and other places in my feature suggestions), I agree it is better to be more general and avoid making assumptions whenever possible.
